### ISSUE TITLE
Basic Database Search Causes Error in MySQL

### ISSUE THREAD

**Issue by MichaelHBBrooks** [Requarks/wiki#842]
### Actual behavior
When performing a search using "Database - Basic" on a MySQL database, the following error message is generated in the browser:
> An expected error occured.

Firefox console log reports:

> [GraphQL error]: Message: select `id`, `title`, `description`, `path`, `localeCode` as `locale` from `pages` where (`isPublished` = true and (title LIKE '%'test'%' COLLATE utf8_general_ci or description LIKE '%'test'%' COLLATE utf8_general_ci)) limit 100 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'test'%' COLLATE utf8_general_ci or description LIKE '%'test'%' COLLATE utf8_gene' at line 1, Location: [object Object], Path: pages,search

### Expected behavior
The search should present zero or more matches with no errors generated.

### Steps to reproduce the behavior
**Operating System:** Ubuntu 18.04.2 LTS
**MySQL Databse version:** Ver 14.14 Distrib 5.7.26
**Wiki JS Versions:** 2.0.0-beta.115 & 2.0.0-beta.91

1. Create a fresh installation of MySQL on the Ubuntu server.
2. Create a `wiki` database.
3. Create a `wikijs` user in `mysql.User` table.
4. Grant privileges to `wikijs` user for the `wiki` database.
5. Perform the steps listed in [the Wiki JS installation guide](https://docs-beta.requarks.io/install/linux) using the `wiki` database and `wikijs` user for the `config.yml` file.
6. Within a web browser, view the Wiki JS server and click on the "Search..." input field at the top center of the page.
7. Type anything.

The following message will appear in red at the top: "An expected error occured." 

### Suggested Fix
There are two errors in the way of the search functionality.
Within `server/modules/search/db/engine.js` are the following two lines around line 42:
```
               builder.whereRaw(`title LIKE '%?%' COLLATE utf8_general_ci`, [q])
               builder.orWhereRaw(`description LIKE '%?%' COLLATE utf8_general_ci`, [q])

```

This is throwing the MySQL search off. If "test" is typed in the search input field in the web page, then the MySQL query will be formatted as:
```title LIKE '%'test'%' COLLATE utf8_general_ci or description LIKE '%'test'%' COLLATE utf8_general_ci```
as seen in the console output.

It should be `'%test%'` instead of `'%'test'%'`.

Fixing that error will then lead to a problem with the collate. It looks like elsewhere the table is defined with `utf8mb4_general_ci` instead of `utf8_general_ci`.

I found the following code change to be valid and resolve the immediate errors, but may require further insight to determine if the collate should be changed here or elsewhere:
```
               builder.whereRaw(`title LIKE '%${q}%' COLLATE utf8mb4_general_ci`)
               builder.orWhereRaw(`description LIKE '%${q}%' COLLATE utf8mb4_general_ci`)
```

**Comment by NGPixel** [comment]
Your proposed fix interpolates the query directly, which can lead to SQL injection. The query must be included as a separate parameter for safety.

I'll do some tests to replicate your error.

**Comment by ndom91** [comment]
I am able to confirm I am receiving the same problem using the 'Database - Basic' provider on wikijs2-230.

I had noticed the `'%'test'%'` incorrect syntax as well, but didnt know where to go from there. Nice bug report @MichaelHBBrooks

**Comment by NGPixel** [comment]
Fixed by d69dde1df5c2db74c10fe86ead50a63528c4487d
